lightly_solution.ipynb

```diff
@@ -331,220 +121,42 @@
-      "[2023-12-19 15:27:24] Writing csv with information about removed samples to 'output_dir/2023-12-19/15:26:00/removed_samples.csv'...\u001b[0m\n",
-      "[2023-12-19 15:27:24] Uploading report...\u001b[0m\n",
-      "[2023-12-19 15:27:27] Done!\u001b[0m\n",
-      "[2023-12-19 15:27:29] Finished compute worker run successfully.\u001b[0m\n",
-      "[2023-12-19 15:27:29] Shutting down...\u001b[0m\n"
+      "[2024-03-26 08:34:15] Lightly Worker Solution v2.11.1\u001b[0m\n",
+      "[2024-03-26 08:34:16] You are using docker build: Tue Mar 12 07:56:29 UTC 2024.\u001b[0m\n",
+      "[2024-03-26 08:34:16] Starting worker with id '65806b455ca68c93b29ad6b3'...\u001b[0m\n",
+      "\u001b[93m[2024-03-26 08:34:16] Worker 2.11.1 can only process jobs scheduled with Lightly Python client 1.5 or higher.\u001b[0m\n",
+      "[2024-03-26 08:34:16] Worker with labels '[]' started. Waiting for jobs...\u001b[0m\n",
+      "[2024-03-26 08:34:16] Found 1 open jobs.\u001b[0m\n",
+      "[2024-03-26 08:34:16] Started job with job_id '6602887ff5e7d0833c925dee'.\u001b[0m\n",
+      "...\n",
+      "...\n",
+      "[2024-03-26 08:35:45] Done!\u001b[0m\n",
+      "[2024-03-26 08:35:48] Finished compute worker run successfully.\u001b[0m\n",
+      "[2024-03-26 08:35:48] Shutting down...\u001b[0m\n"
      ]
     }
    ],
    "source": [
-    "\"\"\"\n",
-    "Run the Lightly Worker to process the run. It mounts the dataset defined earlier.\n",
-    "Ensure that all 3 environment variables are set correctly.\n",
-    "\"\"\"\n",
-    "\n",
-    "!echo \"LIGHTLY_TOKEN=\" $LIGHTLY_TOKEN\n",
-    "!echo \"DATASET_PATH=\" $DATASET_PATH\n",
-    "!echo \"LIGHTLY_WORKER_ID=\" $LIGHTLY_WORKER_ID\n",
-    "\n",
-    "# See if there is another running Lightly Worker that might pick up the job instead.\n",
-    "!docker ps\n",
-    "\n",
-    "if is_nvidia_gpu_available():\n",
-    "  !docker run --shm-size=\"1024m\" --gpus all --rm -it \\\n",
-    "    -v $DATASET_PATH:/input_mount:ro \\\n",
-    "    -v $DATASET_PATH:/lightly_mount \\\n",
-    "    -v $DATASET_PATH:/home/output_dir \\\n",
-    "    -e LIGHTLY_TOKEN=$LIGHTLY_TOKEN \\\n",
-    "    -e LIGHTLY_WORKER_ID=$LIGHTLY_WORKER_ID \\\n",
-    "    lightly/worker:latest\n",
-    "else:\n",
-    "  # Same command, but without `--gpus all` flag.\n",
-    "  !docker run --shm-size=\"1024m\" --rm -it \\\n",
-    "    -v $DATASET_PATH:/input_mount:ro \\\n",
-    "    -v $DATASET_PATH:/lightly_mount \\\n",
-    "    -v $DATASET_PATH:/home/output_dir \\\n",
-    "    -e LIGHTLY_TOKEN=$LIGHTLY_TOKEN \\\n",
-    "    -e LIGHTLY_WORKER_ID=$LIGHTLY_WORKER_ID \\\n",
-    "    lightly/worker:latest\n"
+    "# Run the Lightly Worker to process the scheduled run.\n",
+    "# If you have a Nvidia GPU, add the flag to use it: --gpus all\n",
+    "!docker run --shm-size=\"1024m\" --rm -it \\\n",
+    "    -v {dataset_path}:/input_mount:ro \\\n",
+    "    -v \"./lightly_runs\":/lightly_mount \\\n",
```

Comment:
The mounted directories should be absolute paths. I tried the code and it works on my linux VM but for example on mac docker complains about the paths not being absolute.

Reply:
In this case, I need the pathlib again for creating absolute paths.

Reply:
I decided to use pathlib as follows:

```
# Run the Lightly Worker to process the scheduled run.
# If you have a Nvidia GPU, add the flag to use it: --gpus all

from pathlib import Path
absolute_dataset_path = Path(dataset_path).absolute()
absolute_lightly_path = Path("lightly").absolute()

!docker run --shm-size="1024m" --rm -it \
    -v {absolute_dataset_path}:/input_mount:ro \
    -v {absolute_lightly_path}:/lightly_mount \
    -e LIGHTLY_TOKEN={lightly_token} \
    -e LIGHTLY_WORKER_ID={worker_id} \
    lightly/worker:latest
```

Reply:
It works directly :)